cuBERT_topic_modelling/cuBERTopic.py

```diff
@@ -0,0 +1,563 @@
+from cudf.core.subword_tokenizer import SubwordTokenizer
+import torch
+import cuml
+import cudf
+from cuml.neighbors import NearestNeighbors
+from cuml.manifold import UMAP
+from cuml.metrics import pairwise_distances
+import cupy as cp
+from torch.utils.dlpack import to_dlpack, from_dlpack
+from ctfidf import ClassTFIDF
+from mmr import mmr
+from utils.sparse_matrix_utils import top_n_idx_sparse
+from vectorizer.vectorizer import CountVecWrapper
+import math
+from transformers import AutoModel
+from torch.utils.data import TensorDataset, DataLoader
+import transformers
+from cudf.utils.hash_vocab_utils import hash_vocab
+hash_vocab('vocab.txt', 'voc_hash.txt')
+
+device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+
+class gpu_BERTopic:
+    def __init__(self):
+        self.top_n_words_df = None
+        self.topic_sizes_df = None
+        self.original_topic_mapping = None
+        self.new_topic_mapping = None
+        self.final_topic_mapping = None
+
+    # TODO: find a way to not iterate through the torch.Tensor
+    # using built-in CuPy/cuDF methods
+    def fix_padding(self, tnsr):
+        """Function to fix padding on a torch.Tensor object
+
+        Args:
+            tnsr ([torch.Tensor]): Tensor representing input_ids,
+            attention_mask
+
+        Returns:
+            [torch.Tensor]: trimmed stack of Tensor objects
+        """
+
```

Comment:
This function is not  really needed. Having zeros to the input does not change the computational output of Transformer models. Let's discuss this on a call.

Reply:
Yep that's correct, that it doesn't impact the output of the model, however based on the last call we had you recommended I create this function, to not use unnecessary space (taken by the zeroes), which made sense to me, hence I added this method to preprocess the input before passing it to create the model.

Reply:
TODO: Move this method to the appropriate test file instead, we don't need it for preparing input to the `AutoModel`